### PR TITLE
New ``pkglist`` list manipulation command

### LINKED ISSUE #14144: [feature] Add support for creating MultiPackagesList from graph
### What is your suggestion?

Currently, it's possible to create this object from [graph file](https://github.com/conan-io/conan/blob/release/2.0/conan/api/model.py#L75), however conan graph API returns the [graph object](https://github.com/conan-io/conan/blob/release/2.0/conan/api/subapi/graph.py#L141).

Now, in order to use `MultiPackagesList` to filter the graph and obtain the list of packages that have been built by dependency I need to serialize the graph object into the file and then load it. This is inefficient.

I'm currently working on a custom command that will:
- invoke `conan install --build missing` for each and every build type that we use in the company
- invoke `cmake --preset <generated-preset>` to automatically create the cmake project with installed dependencies

The idea is for this command to also return the list of packages that have been built by dependency, so I need to filter those packages from each and every dependency graph (for each build type), merge everything into a single PackageList, and serialize that into JSON using formatters so that the produced JSON is then compatible with `conan upload` command.

The custom command is intended to be used both by developers (in which case, the JSON pkglist of built dependencies is not important) and CI, where I would like to use the pkglist to upload the built binaries to the Artifactory server, similar to [this example in the Conan documentation](https://docs.conan.io/2/examples/commands/pkglists.html#building-and-uploading-packages).

Currently, I can copy the implementation of the linked code into my custom command, but it uses Conan's private API (in the `conans` package), which I'm not very comfortable using, as even a minor change in the conan version can break it.

Alternative workarounds are also welcome.

### Have you read the CONTRIBUTING guide?

- [ ] I've read the CONTRIBUTING guide

### LINKED ISSUE #14145: [feature] `conan list merge` command
### What is your suggestion?

This is akin to `conan lock merge`.

This is useful when you create a package and immediately build multiple binaries for it, with different configurations (e.g. debug, release, various options) and you want to upload all binaries in a single `conan upload` call so that you benefit from Conan's parallel upload feature.

The flow would be something like this (avoiding multiple `conan create` calls because each call creates a different revision for some reason).

```
conan export . --name pkg --version 1.0

conan install --requires pkg/1.0 -s build_type=Debug --format=json > graph-debug.json
conan install --requires pkg/1.0 -s build_type=Release --format=json > graph-release.json

conan list --graph=graph-debug.json --graph-binaries=build --format=json > built-debug.json
conan list --graph=graph-release.json --graph-binaries=build --format=json > built-release.json

conan list merge --list=built-debug.json --list=built-release.json --list-out=built.json

conan upload -r myremote -c --list=built.json
```

Of course, this can be implemented as a  custom command (but not as `conan list merge` because `list` command is already reserved). However, I think that having this command by default would be a good value.

### PR DESCRIPTION
**Pull request by memsharded**
Changelog: Feature: New ``conan pkglist merge`` command to merge multiple package lists.
Changelog: Feature: New ``conan pkglist find-remote`` command to find matching in remotes for list of packages in the cache.
Docs: https://github.com/conan-io/docs/pull/3704

Close https://github.com/conan-io/conan/issues/14145
Close https://github.com/conan-io/conan/issues/14144

This can be the missing piece for https://github.com/conan-io/conan/pull/15682

### CODE REVIEW

**Review by AbrilRBS — APPROVED** [review]
I want to think of better names if possible, other than that and the todos, this has been a long time coming :)

**Comment by memsharded** [comment]
> I want to think of better names if possible, other than that and the todos, this has been a long time coming :)

Sure, suggestions welcome! I also don't love ``listx`` for example.

**Comment by memsharded** [comment]
Lets rename to ``conan pkglist merge/find-remote``

File: conans/test/integration/command_v2/test_combined_pkglist_flows.py [hunk 1]
```diff
@@ -168,6 +169,90 @@ def test_graph_pkg_list_only_built(self):
         assert len(pkglist["default"]) == 2
 
 
+class TestPkgListFindRemote:
+    """ we can recover a list of remotes for an already installed graph, for metadata download
+    """
+    def test_graph_2_pkg_list_remotes(self):
```
Comment (AbrilRBS) [inline conans/test/integration/command_v2/test_combined_pkglist_flows.py:175]: Maybe 
```
diff --git a/conans/test/integration/command_v2/test_combined_pkglist_flows.py b/conans/test/integration/command_v2/test_combined_pkglist_flows.py
index e6c9ba955..fbd88317d 100644
--- a/conans/test/integration/command_v2/test_combined_pkglist_flows.py
+++ b/conans/test/integration/command_v2/test_combined_pkglist_flows.py
@@ -180,6 +180,7 @@ class TestPkgListFindRemote:
         c.run("create zlib")
         c.run("create app ")
         c.run("upload zlib* -c -r=default")
+        c.run("upload zlib* -c -r=remote2")
         c.run("upload app* -c -r=remote2")
 
         # This install, packages will be in the cache
@@ -197,8 +198,9 @@ class TestPkgListFindRemote:
         assert "Local Cache" not in pkglist
         assert len(pkglist["default"]) == 1
         assert "zlib/1.0" in pkglist["default"]
-        assert len(pkglist["remote2"]) == 1
+        assert len(pkglist["remote2"]) == 2
         assert "app/1.0" in pkglist["remote2"]
+        assert "zlib/1.0" in pkglist["remote2"]
```

to show that it has no problem handling cases where the lib is present in more than one site, even if the user later has to decide where to get the metadata from?

**Review by AbrilRBS — APPROVED** [review]
Minir test suggestion, looks good

File: conans/test/integration/command_v2/test_combined_pkglist_flows.py (hunk 1 above)
Comment (memsharded) [reply conans/test/integration/command_v2/test_combined_pkglist_flows.py:175]: Done

**Review by AbrilRBS — APPROVED** [review]

**Review by czoido — APPROVED** [review]

**Pull request merged**